python/lsst/ts/atwhitelight/config_schema.py

```diff
@@ -0,0 +1,190 @@
+        description: >-
+          Threshold for turning off the lamp and going to fault state
+          due to low supply temperature (C)
+        type: number
+      high_supply_temperature_alarm:
+        description: >-
+          Threshold for turning off the lamp and going to fault state
+          due to high supply temperature (C)
+        type: number
+      low_process_flow_alarm:
+        description: >-
+          Threshold for turning off the lamp and going to fault state
+          due to low process flow (liters/min)
+        type: number
+        minimum: 0
+      connect_timeout:
+        description: >-
+          Maximum time to connect and read watchdog status (seconds).
+          Include time for a few commands to be sent (a few multiples
+          of command_interval), though the code does try to send
+          the read watchdog command first.
+        type: number
+        exclusiveMinimum: 0
+      command_interval:
+        description: >-
+          Minimum interval (seconds) between sending each command to the
+          ThermoTek chiller. The ThermoTek manual claims this must be >= 1
+          but experiments suggest it can be much shorter.
+        type: number
+        exclusiveMinimum: 0
+      telemetry_interval:
+        description: >-
+          Interval (seconds) after one set of telemetry commands
+          finishes before queuing the next set. These commands are run
+          at low priority, so the actual interval before the next command
+          runs may be longer.
+        type: number
+        exclusiveMinimum: 0
+      watchdog_interval:
+        description: >-
+          Minimum interval (seconds) between watchdog commands.
+          Watchdog data is more important than telemetry data;
+          it includes whether the chiller is running and whether
+          there are any alarms or warnings.
+        type: number
+        exclusiveMinimum: 0
+    requiredProperties:
+      - host
+      - port
+      - initial_temperature
+      - low_ambient_temperature_warning
+      - high_ambient_temperature_warning
+      - low_supply_temperature_warning
+      - high_supply_temperature_warning
+      - low_process_flow_warning
+      - low_ambient_temperature_alarm
+      - high_ambient_temperature_alarm
+      - low_supply_temperature_alarm
+      - high_supply_temperature_alarm
+      - low_process_flow_alarm
+      - connect_timeout
+      - command_interval
+      - telemetry_interval
+      - watchdog_interval
+    additionalProperties: false
+  lamp:
+    description: Configuration for the lamp controller and the LabJack that controls it
+    type: object
+    properties:
+      device_type:
+        description: LabJack device type, e.g. T4
+        type: string
+        enum: [T4, T7]
+      connection_type:
+        description: Type of LabJack connection, e.g. TCP
+        type: string
+      identifier:
+        description: LabJack identifier; an IP host address if connection_type=TCP
```

Comment:
is there no port or is it just the standard (21) ?

Reply:
I don't know what port the LabJack ljm library uses, but it is not user-configurable.